File: apps/ios/Pebbles/Features/Profile/Components/ProfileBanner.swift
import SwiftUI

struct ProfileBanner: View {
    let displayName: String?
    let memberSince: Date?
    let glyphStrokes: [GlyphStroke]?

    var body: some View {
        GlyphBanner(
            strokes: glyphStrokes,
            title: displayName ?? "",
            titleFont: .largeTitleHand,
            subtitle: memberSince.map {
                .meta(String(localized: "Member since \($0.formatted(.dateTime.month(.wide).year()))"))
            }
        )
    }
}

#Preview("With glyph (placeholder strokes)") {
    ProfileBanner(displayName: "Alexis", memberSince: Date(), glyphStrokes: nil)
        .padding()
}

#Preview("With glyph") {
    ProfileBanner(
        displayName: "Alexis",
        memberSince: Date(),
        glyphStrokes: [GlyphStroke(d: "M40,40 L160,160", width: 6)]
    )
    .padding()
}

File: apps/android/app/src/main/kotlin/app/pbbls/android/features/profile/components/ProfileBanner.kt
package app.pbbls.android.features.profile.components

import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import app.pbbls.android.R
import app.pbbls.android.features.glyph.models.GlyphStroke
import app.pbbls.android.features.glyph.views.GlyphBanner
import app.pbbls.android.features.glyph.views.GlyphBannerSubtitle
import app.pbbls.android.theme.PebblesTypography
import java.time.OffsetDateTime
import java.time.ZoneId
import java.time.format.DateTimeFormatter
import java.util.Locale

/**
 * Profile header — ports iOS `ProfileBanner.swift`: the [GlyphBanner] with the
 * display name in the hand face and a "Member since <Month Year>" meta line,
 * formatted via the active locale (never pinned — repo rule).
 */
@Composable
fun ProfileBanner(
    displayName: String?,
    memberSince: OffsetDateTime?,
    glyphStrokes: List<GlyphStroke>?,
    modifier: Modifier = Modifier,
) {
    val subtitle =
        memberSince?.let {
            val formatted =
                it
                    .atZoneSameInstant(ZoneId.systemDefault())
                    .format(DateTimeFormatter.ofPattern("MMMM yyyy", Locale.getDefault()))
            GlyphBannerSubtitle.Meta(stringResource(R.string.profile_member_since, formatted))
        }
    GlyphBanner(
        title = displayName.orEmpty(),
        strokes = glyphStrokes,
        titleStyle = PebblesTypography.largeTitleHand,
        subtitle = subtitle,
        modifier = modifier,
    )
}
